client/state/comments/reducer.js

```diff
@@ -193,7 +197,7 @@ export function pendingItems( state = {}, action ) {
 				contiguous: ! action.commentById,
 				has_link: commentHasLink( _comment.content, _comment.has_link ),
 			} ) );
-			const allComments = unionBy( state[ stateKey ], comments, 'ID' );
+			const allComments = unionById( state[ stateKey ], comments );
```

Comment:
There might be a pre-existing bug here. If `comments` has a new version of comment `ID` and `state` has an old version of the same `ID`, the result of the union will contain the old version and the new one will be ignored.

Other usages of `unionBy` changed in this PR do the union in the opposite order: `union( newData, prevState )`.

Reply:
I'm not really sure what the goal was here, which is why I'd prefer to leave this behave the same way it did before. I wonder what's a good way to verify what's the intended behavior 🤔  Happy to follow up in another PR if you are certain this is a bug.